cpp/src/gandiva/exported_funcs_registry.h

```diff
@@ -21,34 +21,31 @@
 #include <vector>
 
 #include <gandiva/engine.h>
+#include <gandiva/visibility.h>
 
 namespace gandiva {
 
 class ExportedFuncsBase;
 
 /// Registry for classes that export functions which can be accessed by
 /// LLVM/IR code.
-class ExportedFuncsRegistry {
+class GANDIVA_EXPORT ExportedFuncsRegistry {
  public:
   using list_type = std::vector<std::shared_ptr<ExportedFuncsBase>>;
 
   // Add functions from all the registered classes to the engine.
   static void AddMappings(Engine* engine);
 
   static bool Register(std::shared_ptr<ExportedFuncsBase> entry) {
-    registered().push_back(entry);
+    registered()->emplace_back(std::move(entry));
     return true;
   }
 
- private:
-  static list_type& registered() {
-    static list_type registered_list;
-    return registered_list;
-  }
+  static list_type* registered();
```

Comment:
Sorry I didn't notice this previously. Is it possible to keep registered() as private and add a new public function
`static const list_type& Registered()`? It will be safer if the mutable instance is not exposed to users and they can still access a const reference.

Reply:
Sure. A new API `static const list_type& Registered()` is added, and the `registered()` API is still private, but I changed the private `registered()` API to return a pinter instead of mutable ref according to Antoine's suggestion above.